app/services/waste_notification.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+class WasteNotification
+  attr_accessor :client, :message_options, :check_date, :waste_types
+
+  def initialize(check_date = Date.today)
+    @check_date = check_date
+  end
+
+  def send_notifications
+    all_waste_registrations = Waste::DeviceRegistration.all
+    # Waste::LocationType
+
+    # Load all possible addresses for registered push notifications
+    possible_addresses = Address.joins(:waste_location_types).where(
+      street: all_waste_registrations.pluck(:street),
+      city: all_waste_registrations.pluck(:city),
+      zip: all_waste_registrations.pluck(:zip)
+    )
+    # possible_addresses
+
+    # Load all PickupTimes of next couple days
+    number_of_days = all_waste_registrations.pluck(:notify_days_before).uniq
```

Comment:
was ist `number_of_days` z.B.? 5? woher kommt die zahl? die zahl wird dann für `dates_to_check` verwendet. ich habe eine ahnung, was da abgeht. aber könnte das bitte an der methode selbst kommentiert werden? welche dates werden in welcher weise da verarbeitet und zurückgegeben?

Reply:
Methode kommentiert